hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -695,5 +690,23 @@ private void onAuthenticationFailed(Address target, ClientConnection connection,
         connectionsInProgress.remove(target);
     }
 
+    private class ClientConnectionChannelErrorHandler implements ChannelErrorHandler {
+        @Override
+        public void onError(Channel channel, Throwable cause) {
+            if (channel == null) {
+                logger.severe(cause);
+            } else {
+                if (cause instanceof OutOfMemoryError) {
+                    logger.severe(cause);
+                }
 
+                ClientConnection connection = (ClientConnection) channel.attributeMap().get(ClientConnection.class);
+                if (cause instanceof EOFException) {
+                    connection.close("Connection closed by the other side", cause);
+                } else {
+                    connection.close("Exception in " + getClass().getSimpleName(), cause);
```

Comment:
Similar to `TcpIpConnectionChannelErrorHandler`, shouldn't use `getClass().getSimpleName()`.